File: 0645-set-mismatch/0645-set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        int dup=-1,missing=-1;
        
        int n=nums.size();
        
        for(int i=0;i<n;i++)
        {
            if(nums[abs(nums[i])-1]<0) dup=abs(nums[i]);
            else 
            {
                nums[abs(nums[i])-1]*=-1;
            }
        }
        
        for(int i=0;i<n;i++)
        {
            if(nums[i]>0) missing = i+1;
        }
        
        return {dup,missing};
    }
};

File: 0645-set-mismatch/0645-set-mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int size = nums.length;
        
        int[] arr = new int[size];
        
        for(int i=0;i<size;i++)
        {
            arr[nums[i]-1]++;
        }
        
        int[] ans = new int[2];
        
        for(int i=0;i<size;i++)
        {
            if(arr[i]==2) ans[0] = i+1;
            else if(arr[i]==0) ans[1] = i+1;
        }
        
        return ans;
    }
}